### ISSUE TITLE
`prefer-export-from` should not report when casting the type

### ISSUE THREAD

**Issue by pcorpet** [sindresorhus/eslint-plugin-unicorn#1579]
When enabling the `prefer-export-from` I get a warning on a valid case: type casting.

For instance in my app I want modules to use my version of redux' useDispatch as I've enforced a different type (narrower).

```js
import {useDispatch as reduxUseDispatch} from 'react-redux'
type MyDispatchType = Dispatch<MyActions>

export const useDispatch: () => DispatchAllActions = reduxUseDispatch
```

As of v38.0.0, eslint would complain on the last line although it's not equivalent to an `export from` statement.

**Comment by fisker** [comment]
#1580
